projects/frontend/cicd/publish_image_dockerhub.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -e
+
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+VDK_DOCKER_REGISTRY_URL=${VDK_DOCKER_REGISTRY_URL:-"registry.hub.docker.com/versatiledatakit"}
+
+[ $# -eq 0 ] && echo "ERROR: No argument for docker image name provided." && exit 1
+[ $# -eq 1 ] && echo "ERROR: No argument for UI project path provided." && exit 1
+[ $# -eq 2 ] && echo "ERROR: No argument for unique patch version provided." && exit 1
+
+name="$1"
+ui_path="$2"
+patch_version="$3"
+
+[ ! -d "$ui_path/dist/ui" ] && echo "ERROR: dist/ui directory does not exist under path $ui_path" && exit 1
+
+version_tag=$(awk -v id=$patch_version 'BEGIN { FS="."; OFS = "."; ORS = "" } { gsub("0",id,$3); print $1, $2, $3 }' $ui_path/version.txt)
+
+image_repo="$VDK_DOCKER_REGISTRY_URL/$name"
+image_tag="$image_repo:$version_tag"
+
+docker build -t "$image_tag" -t "$image_repo:latest" $ui_path
```

Comment:
The publish script actually creates the docker image release candidate? This wouldn't be the production image, since not tested.

The suggestion here is, keep the variety of artifacts (-like cicd test and prod docker images) to a bare minimum needed. All of those should be tested, then the exact **tested artifacts needs to be published** (rather than publishing something partially tested due it is built on top of some priorly tested stack). 
The concept is, to release an atomic artifact that was priorly verified in its entirety.
See https://github.com/vmware/versatile-data-kit/blob/main/projects/frontend/data-pipelines/gui/package.json#L19


1. [build and unit test] release candidates -> NPM artifacts, dist/ 
2. [build docker image] rc with installed release candidate stack
3. [test docker image in cicd env running e2e tests] with installed release candidates
4. [release] all 2.0.0-rc.12345 to 2.0.0

Reply:
@ivakoleva 
What you are suggesting is not possible. 
Conceivably the docker image could be broken if there is something wrong with webpack.
However that exact same problem exists for the control plane image if there was a problem in the jdk.


Issues here are only caught in the heartbeat checks preformed in the vdk playground.


In the very near future we are going to add automated testing to vdk-quickstart. when this happens the docker images will be tested.

Reply:
For having end to end testing, the deliverable artifact being tested, needs to be operational as usual. The rest is e2e tests orchestration or wiring.

Reply:
we aren't testing an artifact in e2e tests. we are testing files in a source directory.